Engines/Wine/QuickScript/Quick Script/script.js

```diff
@@ -1,132 +1,136 @@
-    return this;
-};
-
-/**
- * set executable
- * @param executable executable without path (e.g. "Steam.exe")
- * @param args use array (e.g. ["-applaunch", 409160])
- */
-QuickScript.prototype.executable = function (executable, args) {
-    this._executable = executable;
-    this._executableArgs = typeof args !== 'undefined' ? args : "";
-    return this;
-};
-
-QuickScript.prototype.wineArchitecture = function (wineArchitecture) {
-    this._wineArchitecture = wineArchitecture;
-    return this;
-};
-
-QuickScript.prototype.wineDistribution = function (wineDistribution) {
-    this._wineDistribution = wineDistribution;
-    return this;
-};
-
-QuickScript.prototype.wineVersion = function (wineVersion) {
-    this._wineVersion = wineVersion;
-    return this;
-};
-
-QuickScript.prototype.wineUserSettings = function (wineUserSettings) {
-    // get
-    if (arguments.length == 0) {
-        return this._wineUserSettings;
-    }
-
-    // set
-    this._wineUserSettings = wineUserSettings;
-    return this;
-};
-
-QuickScript.prototype.postInstall = function (postInstall) {
-    this._postInstall = postInstall;
-    return this;
-};
-
-QuickScript.prototype.preInstall = function (preInstall) {
-    this._preInstall = preInstall;
-    return this;
-};
-
-/**
- * creates shortcut
- * @param {string} [prefix] prefix name
- */
-QuickScript.prototype._createShortcut = function (prefix) {
-    var shortcut = new WineShortcut()
-        .name(this._name)
-        .type(this._type)
-        .category(this._category)
-        .prefix(prefix)
-        .search(this._executable)
-        .arguments(this._executableArgs);
-
-    if (this.miniature().isPresent()) {
-        shortcut.miniature(this.miniature().get())
-    }
-    shortcut.create();
-};
+    category(category) {
+        this._category = category;
+        return this;
+    }
+
+    /**
+     * get/set miniature (for the installation and the shortcut)
+     * @param {URI} [miniature] path to the miniature file
+     */
+    miniature(miniature) {
+        // get
+        if (arguments.length == 0) {
+            return this._miniature;
+        }
+
+        // set
+        this._miniature = java.util.Optional.of(miniature);
+        return this;
+    }
+
+    /**
+     * set executable
+     * @param executable executable without path (e.g. "Steam.exe")
+     * @param args use array (e.g. ["-applaunch", 409160])
+     */
+    executable(executable, args) {
+        this._executable = executable;
+        this._executableArgs = typeof args !== 'undefined' ? args : "";
+        return this;
+    }
+
+    wineArchitecture(wineArchitecture) {
+        this._wineArchitecture = wineArchitecture;
+        return this;
+    }
+
+    wineDistribution(wineDistribution) {
+        this._wineDistribution = wineDistribution;
+        return this;
+    }
+
+    wineVersion(wineVersion) {
+        this._wineVersion = wineVersion;
+        return this;
+    }
+
+    wineUserSettings(wineUserSettings) {
```

Comment:
If we use a different name for the fluid API methods, we don't need separate code for the value access anymore:

```javascript
withWineUserSettings(wineUserSettings) {
   this.wineUserSettings = wineUserSettings;
   return this;
}

...
// accessing wineUserSettings
const userSettings = quickScript.wineUserSettings;
```